rules/manager.go

```diff
@@ -416,18 +418,27 @@ func (g *Group) run(ctx context.Context) {
 			case <-g.done:
 				return
 			case <-tick.C:
-				missed := (time.Since(evalTimestamp) / g.interval) - 1
-				if missed > 0 {
-					g.metrics.IterationsMissed.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
-					g.metrics.IterationsScheduled.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
-				}
-				evalTimestamp = evalTimestamp.Add((missed + 1) * g.interval)
-				iter()
+				updateMissedEvalMetrics(g, &evalTimestamp, syncForStateOverrideFunc, iter)
 			}
 		}
 	}
 }
 
+func updateMissedEvalMetrics(g *Group, evalTimestamp *time.Time, syncForStateOverrideFunc SyncForStateOverrideFunc, iterFunc func()) {
```

Comment:
I think the iterFunc also evaluate the rule groups as well. I wonder do we really need to have this into a function?

Can we just put 
```
if syncForStateOverrideFunc != nil {
	syncForStateOverrideFunc(g, time.Now())
}
```
outside?